src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -942,40 +943,45 @@ trait Definitions extends api.StandardDefinitions {
       // look at erased type because we (only) care about what ends up in bytecode
       // (e.g., an alias type is fine as long as is compiles to a single-abstract-method)
       val tpSym: Symbol = erasure.javaErasure(tp).typeSymbol
-
-      if (tpSym.exists && tpSym.isClass && !(tpSym hasFlag (FINAL | SEALED))
+      val info = tp.typeSymbol.info
```

Comment:
I don't see why we couldn't do this -- the symbol and its info should tell us all we need to know. Running test suite locally with `tpSym.info` as RHS.